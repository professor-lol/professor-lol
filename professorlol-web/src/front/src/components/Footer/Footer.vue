<template>
  <footer :class="classList">
    <slot>Footer</slot>
  </footer>
</template>

<script>
export default {
  name: 'AppFooter',
  props: {
    fixed: {
      type: Boolean,
      default: false
    }
  },
  computed: {
    classList () {
      return [
        'app-footer'
      ]
    }
  },
  mounted: function () {
    this.isFixed()
  },
  methods: {
    isFixed () {
      this.fixed ? document.body.classList.add('footer-fixed') : document.body.classList.remove('footer-fixed')
    }
  }
}
</script>
