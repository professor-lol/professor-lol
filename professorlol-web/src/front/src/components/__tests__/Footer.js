import { mount } from "@vue/test-utils";
import Footer from "../Footer/Footer";

describe("Footer.vue", () => {
  // Inspect the raw component options
  it('has isFixed method', () => {
    expect(typeof Footer.methods.isFixed).toBe('function')
  })
  it('renders correctly', () => {
    const wrapper = mount(Footer)
    expect(wrapper.element).toMatchSnapshot()
    expect(wrapper.element.textContent).toEqual('Footer')
    expect(wrapper.classes()).toContain('app-footer')
  })
});
