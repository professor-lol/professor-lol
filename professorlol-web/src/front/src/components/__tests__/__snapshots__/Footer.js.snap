// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer.vue renders correctly 1`] = `
<footer
  class="app-footer"
>
  Footer
</footer>
`;
